src/Datasource/ModelAwareTrait.php

```diff
@@ -97,7 +98,19 @@ public function loadModel($modelClass = null, $modelType = null)
             }
         }
 
-        list(, $alias) = pluginSplit($modelClass, true);
+        $alias = null;
+        $options = [];
+        if (strpos($modelClass, '\\') === false) {
+            list(, $alias) = pluginSplit($modelClass, true);
+        } else {
+            $options['className'] = $modelClass;
+            $alias = substr(
+                $modelClass,
+                strrpos($modelClass, '\\') + 1,
+                -strlen($modelType)
+            );
```

Comment:
Would this be easier to read if it was

```php
$parts = explode('\\', $modelClass);
$alias = array_pop($parts);
```

Reply:
But my code is more efficient with 2 string operations as opposed to 2 array operations :).  Also your example will still need an additional operation to remove the "Table" suffix.